### PR TITLE
SALTO-6407 Update config type to match the current Guide and Themes implementation.

### PR DESCRIPTION
**Pull request by adfineberg**
_Replace me with a description of the changes in this PR_

---

_Additional context for reviewer_

---
_Release Notes_: 
_Replace me with a short sentence that describes the effect of this change on Salto users_

---
_User Notifications_: 
_Replace me with a short sentence that describes changes that will appear in NaCls and are not caused by user actions (e.g. a new annotation, field values that are converted to references, etc). Hidden changes should not be listed._

### CODE REVIEW

File: packages/zendesk-adapter/src/config_creator.ts [hunk 1]
```diff
@@ -56,13 +81,14 @@ export const getConfig = async (options?: InstanceElement): Promise<InstanceElem
   if (options === undefined || !createOptionsTypeGuard<ConfigOptionsType>(optionsElemId)(options)) {
     return defaultConf
   }
-  if (options.value.enableGuide === true || options.value.enableGuideThemes === true) {
+  const { enableGuide, enableGuideThemes, guideOptions } = options.value
+  if (enableGuide === true || enableGuideThemes === true || (guideOptions && guideOptions !== NO_GUIDE)) {
```
Comment (edenhassid) [inline packages/zendesk-adapter/src/config_creator.ts:85]: why do you have `guideOptions` twice?

File: packages/zendesk-adapter/src/config_creator.ts [hunk 2]
```diff
@@ -56,13 +81,14 @@ export const getConfig = async (options?: InstanceElement): Promise<InstanceElem
...
     const configWithGuide = defaultConf.clone()
-    const guideOverride = options.value.enableGuide === true ? { brands: ['.*'] } : {}
```
Comment (edenhassid) [inline packages/zendesk-adapter/src/config_creator.ts:61]: why did you remove this whole logic?

**Review by edenhassid — COMMENTED** [review]
looks good! added 2 small comments

File: packages/zendesk-adapter/src/config_creator.ts (hunk 1 above)
Comment (adfineberg) [reply packages/zendesk-adapter/src/config_creator.ts:85]: because it's optional, if there isn't any `guideOptions` I don't want it to get in this clause

File: packages/zendesk-adapter/src/config_creator.ts (hunk 2 above)
Comment (adfineberg) [reply packages/zendesk-adapter/src/config_creator.ts:61]: I'm going to remove the booleans in a bit, I'm leaving them now for backwards compatibility. 
Currently, the logic is - if any of the booleans is on, then we want guide.

**Review by edenhassid — APPROVED** [review]
